### DISCUSSION TITLE
Regional plugin deletion and retrieval issues.

### DISCUSSION THREAD

**Discussion by lvhaoyu39** [katspaugh/wavesurfer.js#3130] [Q&A] [answered]
Hello, after deleting a certain region, I used the getRegions() method to obtain a list of all regions, including the previously deleted region data.

**Comment by katspaugh** [comment]
What wavesurfer version are you using and how are you deleting a region?

**Reply by lvhaoyu39** [reply]
I am using version 7.1.5
wsRegions.getRegions()
  .filter((region) => region.id === idToDelete)
  .forEach((region) => region.remove())

**Reply by katspaugh** [reply]
Paste the full code please. You're probably doing it in some event handler? Is the region visually removed?

**Reply by lvhaoyu39** [reply]
data.forEach((item, index, arr) => {
    regionList.forEach((region) => {
      if (index !== arr.length - 1 && region.id === item.id) {
        region.remove()
      }
    })
  })
  console.log(wavesurferState.wsRegion.getRegions())

He was visually deleted.

**Reply by lvhaoyu39** [reply]
Hello, I deleted all regions using the clearRegions method, but the previously deleted regions can still be obtained through the getRegions method. Is this a bug?

**Reply by lvhaoyu39** [reply]
wavesurferState.wavesurfer.on('click', () => {
      wavesurferState.wsRegion.clearRegions()
      const regions = wavesurferState.wsRegion.getRegions()
      console.log(regions) 
      // if (regions.length !== 0) {
      //   console.log(regions)
      //   regions.forEach((item) => {
      //     if (item.start !== item.end) {
      //       item.remove()
      //     }
      //   })
      // }
    })

**Reply by katspaugh** [reply]
The code looks correct, so it's probably a bug in the plugin.

**Reply by lvhaoyu39** [reply]
Currently, it seems to be a bug in the plugin

**Reply by lvhaoyu39** [reply]
This issue is currently fatal. When will a version be updated.

**Reply by katspaugh** [reply]
When you make a PR to fix it.

**Reply by lvhaoyu39** [reply]
Sorry, my current level is limited and I may not be able to make any modifications.

**Comment by katspaugh** [comment, marked as answer]
Solved here: https://github.com/katspaugh/wavesurfer.js/discussions/3149#discussioncomment-6851348

> I have solved these two problems and thank you very much for answering them for me. In Vue, reactive or ref cannot be used to declare instances, as this can easily lead to such issues.
